### PR TITLE
[llvm] Avoid 'raw_string_ostream::str()' (NFC)

### PR DESCRIPTION
**Pull request by JOE1994**
Since `raw_string_ostream` doesn't own the string buffer, it is desirable (in terms of memory safety) for users to directly reference the string buffer rather than use `raw_string_ostream::str()`.

Work towards TODO comment to remove `raw_string_ostream::str()`.

### CODE REVIEW

**Comment by JOE1994** [comment]
The following is the TODO comment I'm referring to:

https://github.com/llvm/llvm-project/blob/804415825b97e974c96a92580bcbeaf4c7ff0a04/llvm/include/llvm/Support/raw_ostream.h#L676-L679

File: llvm/include/llvm/Analysis/CFGPrinter.h [hunk 1]
```diff
@@ -148,7 +148,7 @@ std::string CompleteNodeLabelString(
   std::string Str;
   raw_string_ostream OS(Str);
   HandleBasicBlock(OS, *Node);
```
Comment (kazutakahirata) [inline llvm/include/llvm/Analysis/CFGPrinter.h:150]: Could we rename `Str` to `OutStr` and drop the copy construction?

```suggestion
  std::string OutStr;
  raw_string_ostream OS(OutStr);
  HandleBasicBlock(OS, *Node);
```

**Comment by MaskRay** [comment]
LGTM once kazutakahirata's comment is addressed.

File: llvm/include/llvm/Analysis/CFGPrinter.h (hunk 1 above)
Comment (JOE1994) [reply llvm/include/llvm/Analysis/CFGPrinter.h]: Applied your feedback to follow-up commit. Thank you!

**Pull request merged**